src/Microsoft.CodeAnalysis.Razor/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -136,13 +136,17 @@ private ImmutableArray<IParameterSymbol> GetInvokeMethodParameters(INamedTypeSym
 
             if (methods.Length == 0)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_CannotFindMethod(ViewComponentTypes.SyncMethodName, ViewComponentTypes.AsyncMethodName, componentType.ToDisplayString(FullNameTypeDisplayFormat)));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_CannotFindMethod(
+                    ViewComponentTypes.SyncMethodName,
+                    ViewComponentTypes.AsyncMethodName,
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat)));
             }
             else if (methods.Length > 1)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_AmbiguousMethods(componentType.ToDisplayString(FullNameTypeDisplayFormat), ViewComponentTypes.AsyncMethodName, ViewComponentTypes.SyncMethodName));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_AmbiguousMethods(
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat),
+                    ViewComponentTypes.AsyncMethodName,
+                    ViewComponentTypes.SyncMethodName));
```

Comment:
This guy is here to replicate MVC's ViewComponentFactory discovery since we need to provide somewhat understandable errors at design time. Aka, if you have an invalid ViewComponent you'd want to know why you weren't getting TagHelper intellisense for it.